### PR TITLE
Refactor docs and contribution guides

### PR DESCRIPTION
**Pull request by jaggli**
Currently the code examples in community guidelines do not work, directories are wrong and the global scripts in the main monorepo package.json are inconsistent.

- Make scripts consistent in main package.json
- Fix contribution guidelines
- Add link to docs in Readme 
- Separate development guides for `next-yak` and `yak-swc`
- Remove some clutter
- Update pnpm version to v9.12.3
- Rename the contribution guides to `CONTRIBUTION.md`

### CODE REVIEW

File: contributing-guide.md [hunk 1]
```diff
@@ -33,98 +33,129 @@ Before you begin, make sure you have the following installed:
...
+pnpm test:next-yak
+```
 
-For updating test snapshots in the SWC plugin:
+Run `yak-swc` tests:
 
 ```bash
-cd yak_swc
-UPDATE=1 cargo test
+pnpm test:yak-swc
```
Comment (jantimon) [inline contributing-guide.md:96]: that looks wrong

**Review by Mad-Kat — DISMISSED** [review]

**Comment by Mad-Kat** [comment]
Thank you very much for this PR. Our readme, docs and examples need some love. LGTM

File: contributing-guide.md [hunk 2]
```diff
@@ -33,98 +33,129 @@ Before you begin, make sure you have the following installed:
...
+
+```bash
+pnpm test:snapshots:yak-swc
+```
+
 ### Running the Example App
 
-1. Ensure you've built the SWC plugin first:
-   ```bash
-   pnpm run build:swc
```
Comment (jantimon) [inline contributing-guide.md:104]: this step is important - we should not remove it

File: README.md [hunk 3]
```diff
@@ -206,47 +209,45 @@ next-yak converts css-in-js into css modules. This allows to use the full power
...
-          'bg-transparent hover:bg-blue-500 text-blue-700 font-semibold hover:text-white py-2 px-4 border border-blue-500 hover:border-transparent rounded'
+          "bg-transparent hover:bg-blue-500 text-blue-700 font-semibold hover:text-white py-2 px-4 border border-blue-500 hover:border-transparent rounded"
         )}
 `;
 ```
 
-
 ## Build Time Constants
 
-The downside of dynamic properties is that they require inline style attributes.
```
Comment (jantimon) [inline README.md:242]: please keep the line break (`  \n` === `<br>`)

File: README.md [hunk 4]
```diff
@@ -256,59 +257,58 @@ const Container = styled.div`
...
+| File Extension | `.js`, `.jsx`, `.tsx`, etc. | `.yak.js`, `.yak.jsx`, `.yak.tsx`, etc. |
+| Runs at        | Runtime (Node or Browser)   | Compile time (Bundler)                  |
+| Side effects   | ✅                          | 🚫                                      |
+| Yak Features   | All (`styled`, `css`, ...)  | 🚫                                      |
 
 [Build time constants (video)](https://github.com/jantimon/next-yak/assets/4113649/2c83246c-a03b-4c57-8814-32a7248983ac)
 
 ## Yak shaving
 
-While trying to get next-yak to work properly we stumbled accross several bugs.
```
Comment (jantimon) [inline README.md:271]: line break

File: contributing-guide.md [hunk 5]
```diff
@@ -33,98 +33,129 @@ Before you begin, make sure you have the following installed:
...
+
+### Updating Test Snapshots
+
+Update all snapshots
+
 ```bash
 pnpm test:snapshots
 ```
 
+For updating test snapshots in the `next-yak` package:
```
Comment (jantimon) [inline contributing-guide.md:113]: the contribution guide is quite long anyway so I would keep it simple and just keep it to `pnpm test:snapshots`

File: package.json [hunk 6]
```diff
@@ -7,15 +7,20 @@
...
     "example": "pnpm --filter=next-yak-example run dev",
-    "test": "pnpm run build && pnpm --filter=next-yak --filter=cross-file-tests --filter=next-yak-example run \"/test($|:types)/\"",
-    "test:watch": "pnpm --filter=next-yak run test:watch",
-    "test:snapshots": "pnpm --filter=next-yak run test:snapshots",
     "package:types": "npx --package=@arethetypeswrong/cli attw --pack packages/next-yak",
-    "prettier": "pnpm run --recursive --if-present prettier"
+    "prettier": "pnpm run --recursive --if-present prettier",
+    "test": "pnpm run test:next-yak && pnpm run test:swc",
+    "test:snapshots": "pnpm run test:next-yak:snapshots && pnpm run test:swc:snapshots",
+    "test:next-yak": "pnpm run build:next-yak && pnpm --filter=next-yak --filter=cross-file-tests --filter=next-yak-example run \"/test($|:types)/\"",
```
Comment (jantimon) [inline package.json:18]: I like the prettier cmd but please don’t introduce to many scripts - let’s see if we can clean them up a little bit

**Comment by jantimon** [comment]
thanks @jaggli the cleanup was really necessary 👍

**Comment by jantimon** [comment]
we should add

https://github.com/css-modules/postcss-modules-local-by-default/issues/78
and
https://github.com/css-modules/postcss-modules-local-by-default/pull/80

to the yak shaving section

**Review by jantimon — APPROVED** [review]

**Pull request merged**